src/plugins/dashboard/server/content_management/v3/cm_services.ts

```diff
@@ -0,0 +1,544 @@
+  }),
+  useMargins: schema.boolean({
+    defaultValue: true,
+    meta: { description: 'Show margins between panels in the dashboard layout.' },
+  }),
+  syncColors: schema.boolean({
+    defaultValue: true,
+    meta: { description: 'Synchronize colors between related panels in the dashboard.' },
+  }),
+  syncTooltips: schema.boolean({
+    defaultValue: true,
+    meta: { description: 'Synchronize tooltips between related panels in the dashboard.' },
+  }),
+  syncCursor: schema.boolean({
+    defaultValue: true,
+    meta: { description: 'Synchronize cursor position between related panels in the dashboard.' },
+  }),
+});
+
+// These are the attributes that are returned in search results
+export const searchResultsAttributesSchema = schema.object({
+  title: schema.string({ meta: { description: 'A human-readable title for the dashboard' } }),
+  description: schema.string({ defaultValue: '', meta: { description: 'A short description.' } }),
+  timeRestore: schema.boolean({
+    defaultValue: false,
+    meta: { description: 'Whether to restore time upon viewing this dashboard' },
+  }),
+});
+
+export const dashboardAttributesSchema = searchResultsAttributesSchema.extends({
+  // Search
+  kibanaSavedObjectMeta: schema.object(
+    {
+      searchSource: schema.maybe(searchSourceSchema),
+    },
+    {
+      meta: {
+        description: 'A container for various metadata',
+      },
+    }
+  ),
+  // Time
+  timeFrom: schema.maybe(
+    schema.string({ meta: { description: 'An ISO string indicating when to restore time from' } })
+  ),
+  timeTo: schema.maybe(
+    schema.string({ meta: { description: 'An ISO string indicating when to restore time from' } })
+  ),
+  refreshInterval: schema.maybe(
+    schema.object(
+      {
+        pause: schema.boolean({
+          meta: {
+            description:
+              'Whether the refresh interval is set to be paused while viewing the dashboard.',
+          },
+        }),
+        value: schema.number({
+          meta: {
+            description: 'A numeric value indicating refresh frequency in milliseconds.',
+          },
+        }),
+        display: schema.maybe(
+          schema.string({
+            meta: {
+              description:
+                'A human-readable string indicating the refresh frequency. No longer used.',
+              deprecated: true,
+            },
+          })
+        ),
+        section: schema.maybe(
+          schema.number({
+            meta: {
+              description: 'No longer used.', // TODO what is this legacy property?
+              deprecated: true,
+            },
+          })
+        ),
+      },
+      {
+        meta: {
+          description: 'A container for various refresh interval settings',
+        },
+      }
+    )
+  ),
+
+  // Dashboard Content
+  controlGroupInput: schema.maybe(controlGroupInputSchema),
+  panels: schema.arrayOf(panelSchema, { defaultValue: [] }),
```

Comment:
No, it's more involved than that. Below is a table that breaks down the difference between the Dashboard API attributes and Dashboard saved object attributes.

| Saved Object key | Example SO value | Public API key | Example API value | Reason |
| :---- | :---- | :---- | :---- | :---- |
| `panelsJSON` | `“[{\"type\":\"lens\",\"embeddableConfig\":{ … }, … }]”` | `panels` | `[{ “type”:“lens”, “embeddableConfig”: {...}, …}]` | The panelsJSON value is stringified JSON which has no defined schema or validation. The panels value is a JSON object which can be validated against a defined schema. |
| `optionsJSON` | `"{\"useMargins\":true,\"syncColors\":false,\"syncCursor\":true,\"syncTooltips\":false,\"hidePanelTitles\":false}"`  | `options` | `{   	"hidePanelTitles": false,   	"syncColors": false,   	"syncCursor": true,   	"syncTooltips": false,   	"useMargins": true 	}` | As with the panelsJSON property, the optionsJSON property is destringified into a JSON object that is validated against a defined schema. |
| `kibanaSavedObject.searchSourceJSON` | `"{\"query\":{\"language\":\"kuery\",\"query\":\"\"},\"filter\":[]}"`  | `kibanaSavedObject.searchSearch` | `{ "query": {   "language":"kuery", "query": ""  }, "filter": []  }` | Just like the previous \<foo\>JSON properties, the searchSource value is the destringified JSON object that can be validated against a defined schema. |
| `controlGroupInput.panelsJSON` | `"{\"612f8db8-9ba9-41cf-a809-d133fe9b83a8\":{\"grow\":true,\"order\":0,\"type\":\"optionsListControl\", … }”`  | `controlGroupInput.panels` | `[ { "grow": true, "order": 0, "type": "optionsListControl", …  } ]` | Unlike the top-level panelsJSON which is stored as an array of configs, the controlGrouptInput.panelsJSON is stored as an object mapping unique ids to config values.  For the API, we don’t necessarily want consumers to need to define ids for each control when we could do that for them. So the API schema for the controlGroupInput.panels is an array of configs. Specifying an id in the config is optional and a uuid is generated if the id is not specified.  When persisting to the saved object, the array is converted back to the object. |
| `controlGroupInput.ignoreParentSettingsJSON` | `"{\"ignoreFilters\":false,\"ignoreQuery\":false,\"ignoreTimerange\":false,\"ignoreValidations\":false}"` | `controlGroupInput.ignoreParentSettings` | `{ "ignoreFilters": false, "ignoreQuery": false, "ignoreTimerange": false, "ignoreValidations": false }` | The ignoreParentSettings value is the destringified JSON object containing boolean options. This allows the object to be validated against a defined schema. |

Reply:
OK, so it might be wise to surface this in some kind of guide... if you're on a new Kibana it might be best to import the objects then `GET` them from the new endpoints. This could merit a "feature" deprecation in the UA in addition to a docs guide ([example](https://github.com/elastic/kibana/blob/568e40accaaee06a6dadfec2028263889cf7104a/x-pack/plugins/reporting/server/deprecations/reporting_role.ts#L139))